src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex.ExtensionMethodInfo.cs

```diff
@@ -0,0 +1,112 @@
+            /// </summary>
+            public readonly ImmutableArray<int> ComplexExtensionMethodInfo { get; }
+
+            public ExtensionMethodInfo(
+                ImmutableDictionary<string, ImmutableArray<int>> simpleExtensionMethodInfo,
+                ImmutableArray<int> complexExtensionMethodInfo)
+            {
+                SimpleExtensionMethodInfo = simpleExtensionMethodInfo;
+                ComplexExtensionMethodInfo = complexExtensionMethodInfo;
+            }
+
+            public void WriteTo(ObjectWriter writer)
+            {
+                writer.WriteInt32(SimpleExtensionMethodInfo.Count);
+
+                foreach (var kvp in SimpleExtensionMethodInfo)
+                {
+                    writer.WriteString(kvp.Key);
+                    writer.WriteInt32(kvp.Value.Length);
+
+                    foreach (var declaredSymbolInfoIndex in kvp.Value)
+                    {
+                        writer.WriteInt32(declaredSymbolInfoIndex);
+                    }
+                }
+
+                writer.WriteInt32(ComplexExtensionMethodInfo.Length);
+                foreach (var declaredSymbolInfoIndex in ComplexExtensionMethodInfo)
+                {
+                    writer.WriteInt32(declaredSymbolInfoIndex);
+                }
+            }
+
+            public static ExtensionMethodInfo? TryReadFrom(ObjectReader reader)
+            {
+                try
+                {
+                    var simpleExtensionMethodInfo = ImmutableDictionary.CreateBuilder<string, ImmutableArray<int>>();
+                    var count = reader.ReadInt32();
+
+                    for (var i = 0; i < count; ++i)
+                    {
+                        var typeName = reader.ReadString();
+                        var arrayLength = reader.ReadInt32();
+                        var arrayBuilder = ArrayBuilder<int>.GetInstance(arrayLength);
+
+                        for (var j = 0; j < arrayLength; ++j)
+                        {
+                            var declaredSymbolInfoIndex = reader.ReadInt32();
+                            arrayBuilder.Add(declaredSymbolInfoIndex);
+                        }
+
+                        simpleExtensionMethodInfo[typeName] = arrayBuilder.ToImmutableAndFree();
+                    }
+
+                    count = reader.ReadInt32();
+                    var complexExtensionMethodInfo = ArrayBuilder<int>.GetInstance(count);
+                    for (var i = 0; i < count; ++i)
+                    {
+                        var declaredSymbolInfoIndex = reader.ReadInt32();
+                        complexExtensionMethodInfo.Add(declaredSymbolInfoIndex);
+                    }
+
+                    return new ExtensionMethodInfo(simpleExtensionMethodInfo.ToImmutable(), complexExtensionMethodInfo.ToImmutableAndFree());
+                }
+                catch (Exception)
+                {
+                }
```

Comment:
Comment?

Reply:
It seems to be the common case for reading persisted data from disk, if any error occurs, just ignore and abandon it.

Reply:
yeah.  this is consistent.  my bad for coding this way :)